packages/auth/src/grant/service.ts

```diff
@@ -142,12 +146,14 @@ async function rejectGrant(
   })
 }
 
-async function deleteGrant(
+async function revokeGrant(
   deps: ServiceDependencies,
   continueId: string
 ): Promise<boolean> {
-  const deletion = await Grant.query(deps.knex).delete().where({ continueId })
-  if (deletion === 0) {
+  const update = await Grant.query(deps.knex)
+    .patch({ state: GrantState.Revoked })
+    .where({ continueId })
+  if (update === 0) {
```

Comment:
At this point, there are probably still access/accessTokens in the db associated with this revoked grant. Previously they would have been deleted via a cascade with the grant. Just like grants, I am filtering out such `access`/`accessTokens` on the read side but also considered keeping a hard-delete here. I decided against it but I do wonder if these records will accumulated to rather large volumes unnecessarily.

Reply:
I think we should still hard delete here. Revoking a grant is supposed to [revoke all associated access tokens](https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol-15#name-revoking-a-grant-request).

Reply:
I'll change it to hard-delete